### PR TITLE
Move past appointment history/matching clinic check into eligibility checks

### PR DESCRIPTION
**Pull request by narin**
## Description
This PR moves the long term appointment history and matching clinics check into the eligibility check promise chain, and toggles the `directClinics` flag to false if user does not have any past appointment clinics that match.

## Testing done
Local and unit

## Screenshots
image

## Acceptance criteria
- [ ] Error is displayed instead of user getting stuck on past appointment history check

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/actions/newAppointment.js [hunk 1]
```diff
@@ -176,12 +176,11 @@ export function updateFacilityType(facilityType) {
   };
 }
 
-export function startDirectScheduleFlow(appointments) {
+export function startDirectScheduleFlow() {
   recordEvent({ event: 'vaos-direct-path-started' });
 
   return {
     type: START_DIRECT_SCHEDULE_FLOW,
-    appointments,
```
Comment (narin) [inline src/applications/vaos/actions/newAppointment.js:184]: These past appointments get sent to the reducer during the eligibility check instead.

File: src/applications/vaos/tests/newAppointmentFlow.unit.spec.js [hunk 2]
```diff
@@ -342,39 +342,6 @@ describe('VAOS newAppointmentFlow', () => {
 
         resetFetch();
       });
-      it('should be requestDateTime page if past appointments request errors', async () => {
```
Comment (narin) [inline src/applications/vaos/tests/newAppointmentFlow.unit.spec.js:345]: No longer performing the past appointment history fetch here so there shouldn't be any failures to test for.

File: src/applications/vaos/tests/actions/newAppointment.unit.spec.js [hunk 3]
```diff
@@ -674,7 +675,11 @@ describe('VAOS newAppointment actions', () => {
         },
       });
       setFetchJSONResponse(global.fetch.onCall(3), clinics);
-      setFetchJSONResponse(global.fetch.onCall(4), facilityDetails);
+      setFetchJSONResponse(global.fetch.onCall(4), pastAppointments);
```
Comment (narin) [inline src/applications/vaos/tests/actions/newAppointment.unit.spec.js:678]: This is what was tripping me up.  I forgot that the long term history check actually performs 4 fetchs and this stubbing kept causing errors until I repeated them.

File: src/applications/vaos/utils/eligibility.js [hunk 4]
```diff
@@ -68,11 +78,29 @@ export async function getEligibilityData(
   };
 
   if (directData?.length) {
-    const [directPastVisit, clinics] = directData;
+    const [directPastVisit, clinics, pastAppointments] = directData;
+
+    const hasMatchingClinics = clinics?.length
```
Comment (jbalboni) [inline src/applications/vaos/utils/eligibility.js:83]: Is there a test that directly covers this logic? It looks like there might not be, even though it does seem to get inadvertently checked.

**Review by jbalboni — APPROVED** [review]
One question about tests, but LGTM overall.

File: src/applications/vaos/utils/eligibility.js (hunk 4 above)
Comment (narin) [reply src/applications/vaos/utils/eligibility.js:83]: Added tests for these

**Pull request merged**